treeherder/push_health/classification.py

```diff
@@ -0,0 +1,111 @@
+infra = ['raptor-main TEST-UNEXPECTED-FAIL: no raptor test results were found',
+         'Uncaught exception: Traceback (most recent call last):']
+ignore_leaks = ['tab missing output line for total leaks!',
+                'plugin missing output line for total leaks!',
+                'process() called before end of test suite']
+
+ignore = ['[taskcluster:error]',
+          'Return code:',
+          'ABORT',
+          'Received SIGINT (control-C), so stopped run',
+          'Uncaught exception:',
+          '/builds/worker/workspace/mozharness/mozharness/base/script.py',
+          'self.run_action',
+          'self._possibly_run_method',
+          'Automation Error:',
+          'The mochitest suite:',
+          'TBPL FAILURE',
+          'Assertion failure:',
+          'AsyncShutdown timeout',
+          'Force-terminating active process',
+          'InvalidArgumentException:',
+          'JavascriptException:',
+          'ChunkedEncodingError:',
+          'No suite end',
+          'Logged with data',
+          'No tests run',
+          'HTTP Error 500',
+          'runjunit.py',
+          'runtests.py',
+          'No such file or directory',
+          'not a valid structured log message',
+          'timed out loading test page:',  # raptor
+          'Error copying results',
+          'exception output:',
+          'zombiecheck',
+          'shutil error:',
+          'mozdevice.adb.ADBTimeoutError:',
+          'ValueError: invalid literal for int() with base 10:',
+          'timed out after']
+
+
+def set_classifications(failures, intermittent_history, fixed_by_commit_history):
+    for failure in failures:
+        line_messages = get_line_messages(failure)
+        set_infra(failure, line_messages)
+        set_leak(failure, line_messages)
+        set_intermittent(failure, intermittent_history)
+        set_previous_regression(failure, fixed_by_commit_history)
```

Comment:
there is a chance we could hit all of these scenarios based on a set of messages for a given job.  I think for infra/leak these are not test specific and we need to accept these or ignore them.  For intermittent and previous regression, it seems logical.

Reply:
Shoot, I had meant to have it do a ``continue`` if it was successful on any of the functions.  I'll add that.